src/redux/books/books.js

```diff
@@ -2,13 +2,17 @@
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
-const initialState = [];
+const initialState = [
+  { id: 1, title: 'Learn React', author: 'Kent C Doods' },
+  { id: 2, title: 'Do gooder', author: 'Vitor' },
+  { id: 3, title: 'Best bestie', author: 'Karla' },
+];
 
 // Reducer
-export default function bookReducer(state = initialState, action = {}) {
+export default function bookReducer(state = initialState, action) {
   switch (action.type) {
     case ADD_BOOK:
-      return [...state.concat(action.book)];
+      return state.concat(action.book);
```

Comment:
- [ ] Please Ensure that the state is immutable in all reducers (changes are made on a copy of an array).

Reply:
@mikethreels I have ensured that. Concat method does not alter the array. Neither does the spread operator

Reply:
Hi @Nyame-Wolf,
My apologies, you are right, I was looking for the spread operator and because of that I forgot that concat is also not altering the state.